Add auto-heal playbooks

jhernand opened the pull request:
This patch adds a set of playbooks useful for preparing the
auto-heal environment.

To use them create an inventory file like this:

```ini
[nodes]
master.example.com
node0.example.com
node1.example.com

[all:vars]
openshift_autoheal_awx_host = myawx.example.com
openshift_autoheal_awx_admin_password = ...
openshift_autoheal_awx_user_password = ...
```

The `admin` password should be the password of the AWX administrator
user that will be used to create all the relevant objects (the auto-heal
organization, the auto-heal project, the auto-heal user, etc).

The `user` password is the password that will be assigned to the
`autoheal` user that will be created. This is the user that will be used
to launch the auto-heal jobs.

These are the available playbooks, in the order that you will want to
execute them:

- `generate_ssh_keypair.yml`: Generates a new SSH key pair and stores it
  in the `.ssh` directory, in the `id_autoheal` and `id_autoheal.pub`
  files, and copies it to the `authorized_keys` files of all the nodes
  of the cluster.

- `install_awx_cli.yml`: Installs in the local host the AWX CLI.

- `prepare_awx_server.yml`: Prepares the AWX server, creating all the
  required objects.

- `create_example_template.yml`: Creates an example job template with
  a simple playbook that starts the API server.

jhernand committed:
[WIP] Add auto-heal playbooks

This patch adds a set of playbooks useful for preparing the
auto-heal environment.

To use them create an inventory file like this:

```ini
[nodes]
master.example.com
node0.example.com
node1.example.com

[all:vars]
openshift_autoheal_awx_host = myawx.example.com
openshift_autoheal_awx_admin_password = ...
openshift_autoheal_awx_user_password = ...
```

The `admin` password should be the password of the AWX administrator
user that will be used to create all the relevant objects (the auto-heal
organization, the auto-heal project, the auto-heal user, etc).

The `user` password is the password that will be assigned to the
`autoheal` user that will be created. This is the user that will be used
to launch the auto-heal jobs.

These are the available playbooks, in the order that you will want to
execute them:

- `generate_ssh_keypair.yml`: Generates a new SSH key pair and stores it
  in the `.ssh` directory, in the `id_autoheal` and `id_autoheal.pub`
  files, and copies it to the `authorized_keys` files of all the nodes
  of the cluster.

- `install_awx_cli.yml`: Installs in the local host the AWX CLI.

- `prepare_awx_server.yml`: Prepares the AWX server, creating all the
  required objects.

- `create_example_template.yml`: Creates an example job template with
  a simple playbook that starts the API server.

Signed-off-by: Juan Hernandez <jhernand@redhat.com>

jhernand commented:
@ironcladlou, @aweiteka, @cben, @zgalor, @elad661 please review.

ironcladlou commented:
LGTM so far, should I merge so we can continue to iterate and try testing this in INT?

jhernand commented:
Yes, please merge. I removed the WIP prefix. If there is any issue I will fix it in another pull request.

jhernand committed:
Check for master node before starting API server

Currently the auto-heal playbook that is used to start the API server
does so without first checking if the node is a master. This means that
the execution will be marked as failure for non master nodes, and those
nodes will also be marked as failed (inside the AWX database). The ideal
way to solve that would be to define a `masters` group in the inventory,
and run the playbook ony for those nodes. But currently this isn't
feasible because there is no easy way to automatically create an AWX
group and add hosts to it. To work around that this patch modifies the
playbook so that it explicitly checks if the node is a master before
trying to start the service.

Signed-off-by: Juan Hernandez <jhernand@redhat.com>
